Recommend that task configuration is transparent (#524)
ietf-wg-ppm/draft-ietf-ppm-dap: draft-ietf-ppm-dap.md

change parameters or to rotate secret values like collector HPKE configuration
or the VDAF verification key is to configure a new task.

This document does not specify a mechanism for configuring each party with the
task parameters. Such a mechanism SHOULD be transparent so that each party has

> an opportunity to opt out based on the value of any parameter. (See

Author: Reviewer note: I considered refining this to "the value of any _privacy-relevant_ parameter", but I thought it would be a bit bike-sheddy to work out which parameters are privacy-relevant and which aren't.

object storage service, then that cloud vendor would be able to reassemble all
the input shares and defeat privacy.

> ## Transparency of the Task Configuration

Reviewer: ```suggestion
## Task Configuration Agreement and Consistency
```

Reviewer: Transparency is likely to confuse people who may understand it to have the CT or KT-like meaning, which isn't what we're talking about here. Rather, I think what we're talking about is simply agreement.

In order to execute a DAP task, it is necessary for honest parties to ensure
they agree on the configuration of the task. The process by which tasks are
configured is not in scope of this document. This means that, in general, it is
possible for a party to participate in a task without knowing all of the
relevant parameters. For example, a Client can upload a report
({{upload-flow}}) without knowing the minimum batch size that is enforced by
the Aggregators during collection ({{collect-flow}}).

> The task configuration mechanism SHOULD ensure that all task parameters are
> visible to all parties such that each party can choose whether to participate
> in the task based on the value of any parameter. This includes the parameters
> enumerated in {{task-configuration}} and any additional parameters implied by
> upload extensions {{upload-extensions}} used by the task.

Reviewer: ```suggestion
Depending on the deployment model, agreement can require that task parameters
are visible to all parties such that each party can choose whether to participate
based on the value of any parameter. This includes the parameters enumerated
in {{task-configuration}} and any additional parameters implied by upload
extensions {{upload-extensions}} used by the task. It also includes the task itself:
meaningful privacy requires that multiple clients contribute to a task and therefore
share a consistent view of the task configuration.
```

object storage service, then that cloud vendor would be able to reassemble all
the input shares and defeat privacy.

## Task Configuration Agreement and Consistency

In order to execute a DAP task, it is necessary for honest parties to ensure
they agree on the configuration of the task. The process by which tasks are
configured is not in scope of this document. This means that, in general, it is
possible for a party to participate in a task without knowing all of the
relevant parameters. For example, a Client can upload a report
({{upload-flow}}) without knowing the minimum batch size that is enforced by
the Aggregators during collection ({{collect-flow}}).

Depending on the deployment model, agreement can require that task parameters
are visible to all parties such that each party can choose whether to participate
based on the value of any parameter. This includes the parameters enumerated
in {{task-configuration}} and any additional parameters implied by upload
extensions {{upload-extensions}} used by the task. It also includes the task itself:
meaningful privacy requires that multiple clients contribute to a task and therefore

> share a consistent view of the task configuration.

Reviewer 2: This is confusing: what is a task but the parameters that constitute it? I think the text you had before was better, though I like that this version removes the all-caps SHOULD.

Reviewer: It's the set of all parameters. Can you offer suggestions?

Reviewer 2: ```suggestion
based on the value of any parameter. This includes the parameters enumerated
in {{task-configuration}} and any additional parameters implied by upload
extensions {{upload-extensions}} used by the task. Since meaningful privacy
requires that multiple clients contribute to a task, they should also share a
consistent view of the task configuration.
```
